### DISCUSSION TITLE
Downloading multiple variables to one file using FastHerbie

### DISCUSSION THREAD

**Discussion by agentrrb** [blaylockbk/Herbie#137] [General]
Hi there,

Thanks so much for sharing your work - open science and data access is paramount.

I'd like to use Herbie to download HRRR data and I only need 8 variables:
1. Air Temperature [K] (TMP:2 m), 2. Relative Humidity [%] (RH:2 m), 3. U-component of Wind [m/s] (UGRD:10 m), 4. V-component of Wind [m/s] (VGRD:10 m), 5. Total Precipitation [mm or kg/m^2] (APCP: surface), 6. Downward Shortwave Radiation Flux [W/m^2] (DSWRF: surface), 7. Elevation [m] (HGT: surface), 8. Total Cloud Cover [%] (TCDC: entire atmosphere)

It seems like FastHerbie with a search string for subsetting is the way to go. Here's a section of code that I'm using, developed from the tutorials you have posted:

`#Create a range of dates
#Pull only a handful of hours as a test
DATES = pd.date_range('2019-01-01 00:00', '2019-01-01 02:00', freq='1H')

#Create a range of forecast lead times
fxx = range(0,4)

#Make FastHerbie Object.
FH = FastHerbie(DATES, model='hrrr', fxx=fxx)

#Subset the GRIB2 files and download
#Search String information: https://blaylockbk.github.io/Herbie/_build/html/user_guide/searchString.html

FH.download(":(TMP:2 m|RH:2 m|UGRD:10 m|VGRD:10 m|APCP:surface|HGT:surface|TCDC:entire atmosphere):")`

The code seems to work, but there are four files that are saved per hour iteration. I think it's saving one file for the 2m data, one for the 10 m data, one for the surface data, and one for the entire atmosphere data. I've provided a screenshot of my working directory in case that's helpful. I think I need all of those variables saved into one grib file for each hourly interval. Is this possible?
![image](https://user-images.githubusercontent.com/68090958/211130641-58b21725-95fc-4abe-a63a-778a8ed04d68.png)

Thanks!!

**Comment by agentrrb** [comment]
Hi there,

I realize now that this is downloading forecast lead times, 4 per hour, rather than creating different files for different variable sets.

**Comment by scollis** [comment]
I'm going to add my question here. 
I had some nice success with FH and temp/dp
```python
mt = "2023-01-15 12:00"

FH = FastHerbie([mt], model="hrrr", fxx=range(2,47))
FH.download("(TMP|RH):2 m") #This needs a good (and cheap) internet connection... ~1m on my connection
ds = FH.xarray("(TMP|RH):2 m", remove_grib=True)
```
gives a nice forecast time series. 

When I want to dig into precip its another story
```mt = "2023-01-15 12:00"

FH = FastHerbie([mt], model="hrrr", fxx=range(2,47))

FH.download(":APCP:surface:[1-9]*-[1-9]*")
```

can not find the field.

however
```python
rflist = []
for fchr in range(2,15):
    rf = Herbie(mt, fxx=fchr).xarray(":APCP:surface:[1-9]*-[1-9]*")
    rflist.append(rf)

rf_ds = xr.concat(rflist, 'step')
```

works.. but only up to forecast hour 11

Is this because one hour precip rates are only available for FH 2-11?

**Reply by scollis** [reply]
ok ... looks like I need to write a better search string

**Comment by scollis** [comment]
ok! I brute forced it 😄 
```
qstr = '(' 
nfcst = 10
for i in range(nfcst):
    if i != nfcst -1:
        qstr = qstr +  f':APCP:surface:{i}-{i+1}*|'
    else:
        qstr = qstr +  f':APCP:surface:{i}-{i+1}*)'
    ```

**Comment by scollis** [comment]
https://github.com/scollis/notebooks/blob/master/Herbie_Chicago_precip.ipynb

I'll open an issue..
